src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,368 @@
+            locationInfo.locationInfo.SetNonNull(Structs::HomeLocationStruct::Type());
+
+            locationInfo.locationInfo.Value().floorNumber = aFloorNumber;
+            locationInfo.locationInfo.Value().areaType    = aAreaType;
+        }
+        else
+        {
+            locationInfo.locationInfo.SetNull();
+        }
+
+        locationInfo.landmarkTag = aLandmarkTag;
+        locationInfo.positionTag = aPositionTag;
+        locationInfo.surfaceTag  = aSurfaceTag;
+
+        // Save the location name to buffer, so access and lifetime is controlled.
+        // If aLocationName is not null, create a CharSpan to reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
+        memset(mLocationNameBuffer, 0, sizeof(mLocationNameBuffer + 1));
+
+        // this assumes locationInfo structure was created above, if appropriate
+        if (!locationInfo.locationInfo.IsNull())
+        {
+            if (aLocationName.empty())
+            {
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, 0);
+            }
+            else if (aLocationName.size() > sizeof(mLocationNameBuffer))
+            {
+                // Save the truncated name that fits into available size.
+                memcpy(mLocationNameBuffer, aLocationName.data(), sizeof(mLocationNameBuffer));
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, sizeof(mLocationNameBuffer));
+            }
+            else
+            {
+                // Save full name.
+                memcpy(mLocationNameBuffer, aLocationName.data(), aLocationName.size());
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, aLocationName.size());
+            }
+        }
+    }
+
+    /**
+     * @brief Compare the location's name with the given text.
+     * @param[in] aLocationName The name to compare.
+     * @return true if the location structure's name field matches aLocationName.
+     *
+     * @note if locations structure's name field is null, returns false.
+     */
+    bool DoesNameMatch(const CharSpan & aLocationName) const
+    {
+        if (!locationInfo.locationInfo.IsNull())
+        {
+            return locationInfo.locationInfo->locationName.data_equal(aLocationName);
+        }
+
+        return false; // name is Null or text sizes do not match
+    }
+
+    /**
+     * @brief Checks if the given LocationStructureWrapper is equal to this one.
+     * @param aOther The location to compare with.
+     * @param aIgnoreLocationId If true, the location IDs are ignored when checking for equality.
+     * @param aIgnoreMapId If true, the map IDs are ignored when checking for equality.
+     * @return True if both locations are equal. False otherwise.
+     */
+    bool IsEqual(const LocationStructureWrapper & aOther, bool aIgnoreLocationId, bool aIgnoreMapId) const
```

Comment:
do not provide several bool parameters to methods. You could use a bitmask or other methods. Otherwise `IsEqual(other, true, false)` is not actually readable and it forces everyone to assign names in comments or other workarounds.